examples/policies/private-token-policy/private-token-policy-v1.pact

```diff
@@ -0,0 +1,140 @@
+(namespace (read-msg 'ns))
+
+(module private-token-policy-v1 GOVERNANCE
+
+  (defconst ADMIN-KS:string "marmalade-examples.private-token-policy")
+
+  (defcap GOVERNANCE ()
+    (enforce-guard ADMIN-KS))
+
+  (implements kip.token-policy-v2)
+  (implements kip.updatable-uri-policy-v1)
+  (use kip.token-policy-v2 [token-info])
+  (use marmalade-v2.guard-policy-v1 [URI-GUARD-MSG-KEY])
+
+  (defschema revealed-tokens-schema
+    revealed:bool  
+  )
+
+  (deftable revealed-tokens:{revealed-tokens-schema})
+
+  (defcap TOKEN_REVEALED (token-id:string uri:string)
+    @doc "Emitted when the token URI has been revealed"
+    @event
+    true
+  )
+
+  (defun has-guard-policy:bool (policies)
+    (> (length (filter (lambda (policy) (= policy marmalade-v2.guard-policy-v1)) policies)) 0))
```

Comment:
You should stringify the `marmalade-v2.guard-policy-v1` because the modref comparison depends on its version (hash).

Reply:
We have a function specifically built to check this for concrete-policies: 
https://github.com/kadena-io/marmalade/blob/main/pact/marmalade-util/util-v1.pact#L79:L86